core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -96,7 +96,16 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (!config.containsMacro("timeTable") && config.timeTable != null) {
       pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
     }
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    if (config.getSchema() != null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    } else {
+      if (config.format.equalsIgnoreCase("text")) {
```

Comment:
you can have one layer of ifs instead of having a nested if.

```
if (config.getSchema() != null) {
...
} else if ("text".equalsIgnoreCase(config.format)) {
...
} else {
...
}